nextcord/components.py

```diff
@@ -202,7 +208,59 @@ def to_dict(self) -> ButtonComponentPayload:
         return payload  # type: ignore
 
 
-class SelectMenu(Component):
+class SelectMenuBase(Component):
+    """Represents a Discord Bot UI Kit Select Menu.
+
+    This is the base class for all select menus.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    custom_id: :class:`str`
+        The ID of the select menu that gets received during an interaction.
+    disabled: :class:`bool`
+        Whether the select menu is disabled or not.
+    placeholder: Optional[:class:`str`]
+        The placeholder of the select menu, if any.
+    min_values: :class:`int`
+        The minimum number of values that must be chosen.
+    max_values: :class:`int`
+        The maximum number of values that can be chosen.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "custom_id",
+        "disabled",
+        "placeholder",
+        "min_values",
+        "max_values",
+    )
+
+    __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
+
+    def __init__(self, data: SelectMenuBasePayload):
+        self.custom_id: str = data["custom_id"]
+        self.disabled: bool = data.get("disabled", False)
+        self.placeholder: Optional[str] = data.get("placeholder")
+        self.min_values: int = data.get("min_values", 1)
+        self.max_values: int = data.get("max_values", 1)
+
+    def to_dict(self) -> SelectMenuBasePayload:
+        payload: SelectMenuBasePayload = {
+            "custom_id": self.custom_id,
+            "disabled": self.disabled,
+            "min_values": self.min_values,
+            "max_values": self.max_values,
+        }
+
+        if self.placeholder:
+            payload["placeholder"] = self.placeholder
+
+        return payload
+
+
+class SelectMenu(SelectMenuBase):
```

Comment:
This should be renamed to `StringSelectMenu` with an alias called `SelectMenu`, like done in `ui.select`.

Reply:
could I also make just an alias for SelectMenu instead of renaming that class? The original Select was also not renamed, StringSelect is the alias.